### ISSUE TITLE
Investigate potential upgradeability issues in Application Connector

### ISSUE THREAD

**Issue by akgalwas** [kyma-project/kyma#2397]
**Description**

We need to make sure that Application Connector components upgrade works correctly.
Test scenario:
- deploy 0.6.1 release to OnDemand cluster
- create Applications, generate client certificate, register services, deploy lambda functions (mind both scenarios: lambda triggered by event, lambda calling registered API)
- build new images for Application Connector components from master branch
- upgrade Application Connector components according to the upgrade instructions 
- perform tests:
   - make sure that services registered before upgrade are intact and deployed lambdas work correctly
   - run acceptance tests and integration test  
- log issues found

**Reasons**

Starting from 0.7 release we need to support upgradability. We need to make sure the upgrade process will work in case of non-breaking changes.

**Comment by Maladie** [comment]
I have called helm upgrade on a cluster with 0.6.1 Kyma installed with lambdas set up (two lambdas triggered by events and two calling APIs). The upgrade replaced all Application Connector-related images to the most recent ones. After that I have performed the following checks:
- lambdas triggered by events
- lambdas calling APIs
- new services created
- acceptance tests run
- manual registration of a new lambda (with new application, service etc.)
- e2e test with the use of application-connector-e2e Groovy application.

**Conclusion**
All checks passed, therefore no issues were found.
